src/Component/GridRegistry.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid;
+
+final class GridRegistry
+{
+    private array $grids;
+
+    public function __construct(\Traversable $grids)
+    {
+        $this->grids = iterator_to_array($grids);
```

Comment:
What's wrong with storing the iterator here?

Reply:
To make that possible
```php
return $this->grids[$code] ?? null;
```
I just followed the [Symfony documentation](https://symfony.com/doc/current/service_container/tags.html#tagged-services-with-index)

Reply:
Ah yeah that makes sense.

Reply:
Well, as this class is part of the component namespace I think it shouldn't rely on Symfony's `tagged_iterator` behavior. Instead it should only accept an array of `GridInterface` and manually index it by `getName`. The Bundle needs to implement a compiler pass.

Reply:
@smoench I think you're right. I improved the GridRegistry to be detached tagged_iterator (with a map) behaviour.